bookOfHb.md

```diff
@@ -0,0 +1,400 @@
+# The Book of Self-Repair (Heartbeat First Impressions)
+
+##  Overall Goals of Self-Repair
+
+1.  Host self-repair recipes, outside of firefox core.  Run them with Firefox permissions.  Allow fast development off the trains.
+
+2.  Collect randomly sampled User Sentiment scores.  Implemented as "Heartbeat First Impressions" recipe.
+
+
+## As a developer, I want to successfully hack on Self-Repair
+
+### Setup (do once)
+
+1.  Zen Of Cooking
+
+	This is the short version. `Cmd+F` will find bigger explanations of nearly everything later on in this document.  
+
+   Mastering this may take several attempts.  Skip around.  Be gentle on yourself!  Enjoy the process!
+   
+1. Pots and Pans
+
+	```
+	# required: git, node
+	brew install node   # includes npm
+	brew install git    # git
+	
+	# optional:  hub, git
+	brew install hub    # github's replacement git client
+	brew install ghi    # github issues client
+	```
+
+1. Groceries
+
+	```
+	# usual git project clone and fork
+	
+	git clone mozilla/self-repair-server
+	cd self-repair-server
+	hub fork
+	```
+
+1. Prep the veggies
+
+	```
+	# usual node project package install
+	npm install   
```

Comment:
When I do this, I get this error:

```
npm ERR! TypeError: Object.keys called on non-object
npm ERR!     at Function.keys (native)
npm ERR!     at installTargetsError (/usr/lib/node_modules/npm/lib/cache.js:708:24)
npm ERR!     at /usr/lib/node_modules/npm/lib/cache.js:638:10
npm ERR!     at saved (/usr/lib/node_modules/npm-registry-client/lib/get.js:148:7)
npm ERR!     at /usr/lib/node_modules/graceful-fs/polyfills.js:133:7
npm ERR!     at Object.oncomplete (fs.js:108:15)
npm ERR! If you need help, you may report this log at:
npm ERR!     <http://github.com/isaacs/npm/issues>
npm ERR! or email it to:
npm ERR!     <npm-@googlegroups.com>

npm ERR! System Linux 4.1.6-201.fc22.x86_64
npm ERR! command "node" "/usr/bin/npm" "install"
npm ERR! cwd /home/willkg/mozilla/self-repair-server
npm ERR! node -v v0.10.36
npm ERR! npm -v 1.3.6
npm ERR! type called_on_non_object
npm http 304 https://registry.npmjs.org/karma-webpack
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/willkg/mozilla/self-repair-server/npm-debug.log
npm ERR! not ok code 0
```

First off, that's pretty high in my "pretty fucking useless" list. Second, I think it's saying the version of webpack specified in `package.json` isn't available on npm.

If I change it to the current version which is 0.12.2, then it dies similarly, but this time on openr. If I change that version to 1.4.1, then it dies on thirdparty. If I remove that, it works fine.

I've got an older version of node.js and npm--maybe that's the issue?